ivy/functional/frontends/tensorflow/math.py

```diff
@@ -549,6 +549,22 @@ def tanh(x, name=None):
 def rsqrt(x, name=None):
     return ivy.reciprocal(ivy.sqrt(x))
 
+@to_ivy_arrays_and_back
+def segment_mean(
+    data, segment_ids, name="segment_mean"
+):
+    ivy.utils.assertions.check_equal(
+        list(segment_ids.shape), [list(data.shape)[0]], as_array=False
+    )
+    x = ivy.zeros(tuple([segment_ids[-1] + 1] + (list(data.shape))[1:]))
+    count = ivy.zeros((segment_ids[-1] + 1,))
+    for i in range((segment_ids).shape[0]):
```

Comment:
Can you investigate weather this could be done using the ivy functional API instead of for loops. I haven't given it much thought, but can you not use ivy.mean to calculate the mean of each segment?